tests/unit/middlewares/withAuth.test.ts

```diff
@@ -0,0 +1,261 @@
+import { Socket } from "net";
+
+import * as refreshTokensFlow from "@lib/commands/refreshTokensFlow";
+import { ExtendedRequest } from "@src/@types/core/ExtendedRequest";
+import { Handler } from "@src/@types/core/Handler";
+import * as getAndPutUser from "@src/commands/getAndPutUser";
+import { config } from "@src/config";
+import { withAuth } from "@src/middlewares/withAuth";
+import { withSession } from "@src/middlewares/withSession";
+import { wrapMiddlewares } from "@src/middlewares/wrapper";
+import * as getDBUserFromSub from "@src/queries/getDBUserFromSub.ts";
+import * as decryptVerifyAccessToken from "@src/workflows/decryptVerifyAccessToken";
+
+const spiedOnDecryptVerifyAccessToken = jest
+  .spyOn(decryptVerifyAccessToken, "decryptVerifyAccessToken")
+  .mockImplementation(async () => {
+    return defaultPayload;
+  });
+
+const spiedOnGetDBUserFromSub = jest
+  .spyOn(getDBUserFromSub, "getDBUserFromSub")
+  .mockImplementation(async () => {
+    return {
+      sub: defaultPayload.sub,
+      refresh_token: "refresh_token",
+      temporary_identities: [],
+    };
+  });
+
+const spiedOnRefreshTokensFlow = jest
+  .spyOn(refreshTokensFlow, "refreshTokensFlow")
+  .mockImplementation(async () => {
+    return {
+      refresh_token: "new_refresh_token",
+      access_token: "new_access_token",
+    };
+  });
+
+const spiedOnGetAndPutUser = jest
+  .spyOn(getAndPutUser, "getAndPutUser")
+  .mockImplementation(async () => {
+    return;
+  });
+
+let mockedVerifyJWTCall = 0;
+
+jest.mock("@src/config", () => {
+  return {
+    config: {
+      connectAccountSessionSalt: "#*b+x3ZXE3-h[E+Q5YC5`jr~y%CA~R-[",
+      connectOpenIdConfigurationUrl: "",
+      connectApplicationClientId: "",
+      connectApplicationClientSecret: "c9ab0fdc-b2dc-47ad-933b-87cf1b180ab5",
+      connectAccountRedirectUri: "",
+      connectAudience: "",
+      connectApplicationScopes: "",
+      connectProviderUrl: "http://foo.test",
+      connectAccountURL: "http://foo.test",
+      dynamoRegion: "eu-west-3",
+    },
+    oauth2Client: {
+      verifyJWT: async () => {
+        mockedVerifyJWTCall = +1;
+
+        return {
+          sub: "2a14bdd2-3628-4912-a76e-fd514b5c27a8",
+        };
+      },
+    },
+  };
+});
+
+async function sealJWS(access_token: string, salt?: string): Promise<string> {
+  return seal(
+    {
+      persistent: {
+        "user-session": {
+          access_token,
+          sub: "2a14bdd2-3628-4912-a76e-fd514b5c27a8",
+        },
+      },
+      flash: {},
+    },
+    salt || config.connectAccountSessionSalt,
+    defaults,
+  );
+}
+
+function mockedReqAndRes(
+  sealedJWE: string,
+): {
+  mockedExtendedRequest: ExtendedRequest;
+  mockedResponse: ServerResponse;
+} {
+  const mockedRequest = new IncomingMessage(new Socket());
+  const mockedResponse = new ServerResponse(mockedRequest);
+
+  const mockedSession = {
+    set: () => {
+      return;
+    },
+    get: () => {
+      return undefined;
```

Comment:
```suggestion
      return;
```

Reply:
This needs to return `undefined` to follow the type.

Reply:
👍